backend/drm/renderer.c

```diff
@@ -178,47 +178,35 @@ static void free_eglimage(struct gbm_bo *bo, void *data) {
 static struct wlr_texture *get_tex_for_bo(struct wlr_drm_renderer *renderer,
 		struct gbm_bo *bo) {
 	struct tex *tex = gbm_bo_get_user_data(bo);
-	if (tex) {
+	if (tex != NULL) {
 		return tex->tex;
 	}
 
-	// TODO: use wlr_texture_upload_dmabuf instead
-
-	tex = malloc(sizeof(*tex));
-	if (!tex) {
-		wlr_log_errno(L_ERROR, "Allocation failed");
+	tex = calloc(1, sizeof(struct tex));
+	if (tex == NULL) {
 		return NULL;
 	}
 
-	tex->egl = &renderer->egl;
-
-	int dmabuf_fd = gbm_bo_get_fd(bo);
-	uint32_t width = gbm_bo_get_width(bo);
-	uint32_t height = gbm_bo_get_height(bo);
-
-	EGLint attribs[] = {
-		EGL_WIDTH, width,
-		EGL_HEIGHT, height,
-		EGL_LINUX_DRM_FOURCC_EXT, gbm_bo_get_format(bo),
-		EGL_DMA_BUF_PLANE0_FD_EXT, dmabuf_fd,
-		EGL_DMA_BUF_PLANE0_OFFSET_EXT, gbm_bo_get_offset(bo, 0),
-		EGL_DMA_BUF_PLANE0_PITCH_EXT, gbm_bo_get_stride_for_plane(bo, 0),
-		EGL_IMAGE_PRESERVED_KHR, EGL_FALSE,
-		EGL_NONE,
+	struct wlr_dmabuf_buffer_attribs attribs = {
+		.n_planes = 1,
+		.width = gbm_bo_get_width(bo),
+		.height = gbm_bo_get_height(bo),
+		.format = gbm_bo_get_format(bo),
 	};
+	attribs.offset[0] = 0;
+	attribs.stride[0] = gbm_bo_get_stride_for_plane(bo, 0);
+	attribs.modifier[0] = 0;
+	attribs.fd[0] = gbm_bo_get_fd(bo);
 
-	tex->img = eglCreateImageKHR(renderer->egl.display, EGL_NO_CONTEXT,
-		EGL_LINUX_DMA_BUF_EXT, NULL, attribs);
-	if (!tex->img) {
-		wlr_log(L_ERROR, "Failed to create EGL image");
-		abort();
-	}
+	// TODO: add EGL_IMAGE_PRESERVED_KHR, EGL_FALSE to attribs?
```

Comment:
It's a very minor optimisation thing. But even then, I think it defaults to false anyway.
I think I originally put it there, because I was experimenting what different it made setting it to true.